services/gateway/app.js

```diff
@@ -51,117 +51,145 @@ const loggerConf = {
 LoggerConfig(loggerConf);
 
 const logger = Logger();
+const MODULE = {
+	DYNAMIC_REWARD: 'dynamicReward',
+	REWARD: 'reward',
+};
 
-const broker = Microservice({
+const defaultBrokerConfig = {
 	name: 'gateway',
 	transporter: config.transporter,
 	brokerTimeout: config.brokerTimeout, // in seconds
 	logger: loggerConf,
-}).getBroker();
-
-const sendSocketIoEvent = (eventName, payload) => {
-	broker.call('gateway.broadcast', {
-		namespace: '/blockchain',
-		event: eventName,
-		args: [payload],
-	});
 };
 
-const gatewayConfig = {
-	transporter: config.transporter,
-	mixins: [ApiService, SocketIOService],
-	name: 'gateway',
-	actions: {
-		spec(ctx) { return genDocs(ctx); },
-		status() { return getStatus(this.broker); },
-		ready() { return getReady(this.broker); },
-		isBlockchainIndexReady() { return getIndexStatus(); },
-	},
-	settings: {
-		host,
-		port,
-		path: '/api',
-		etag: 'strong',
-		use: [],
-
-		cors: {
-			origin: '*',
-			methods: ['GET', 'OPTIONS', 'POST', 'PUT', 'DELETE'],
-			allowedHeaders: [
-				'Content-Type',
-				'Access-Control-Request-Method',
-				'Access-Control-Request-Headers',
-				'Access-Control-Max-Age',
-			],
-			exposedHeaders: [],
-			credentials: false,
-			maxAge: 3600,
-		},
+// Use temporary service to fetch registered sdk modules
+const tempApp = Microservice({
+	...defaultBrokerConfig,
+	name: 'temp_service_gateway',
+	events: {},
+	dependencies: [
+		'connector',
+	],
+});
+
+tempApp.run().then(async () => {
+	// Prepare routes
+	const { modules: registeredModules } = await tempApp.requestRpc('connector.getSystemMetadata');
+	const registeredModuleNames = registeredModules.map(
+		module => module.name === MODULE.REWARD ? MODULE.DYNAMIC_REWARD : module.name,
+	);
```

Comment:
Same as above.
Also, it'd be nice if the code implementation at both the places are consistent.

Reply:
Addressed.